crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -71,8 +80,30 @@ pub type BoxedCondition<In = ()> = Box<dyn ReadOnlySystem<In = In, Out = bool>>;
 /// # world.insert_resource(DidRun(false));
 /// # app.run(&mut world);
 /// # assert!(world.resource::<DidRun>().0);
-pub trait SystemCondition<Marker, In: SystemInput = ()>:
-    sealed::SystemCondition<Marker, In>
+/// ```
+///
+/// A condition returning a `Result<(), BevyError>`
+///
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// # #[derive(Component)] struct Foo;
+/// fn single_foo(q_foo: Query<(), With<Foo>>) -> Result {
```

Comment:
I'd prefer to avoid things like `Foo` in our docs: it makes it harder for users to see the value in these tools. How about a `player_exists` run condition?